lib/puppet/functions/unwrap.rb

```diff
@@ -0,0 +1,39 @@
+# Unwraps a Sensitive value and returns the wrapped object.
+#
+# ~~~puppet
+# $plaintext = 'hunter2'
+# $pw = Sensitive.new($plaintext)
+# notice("Wrapped object is $pw") #=> Prints "Wrapped object is Sensitive [value redacted]"
+# $unwrapped = $pw.unwrap
+# notice("Unwrapped object is $unwrapped") #=> Prints "Unwrapped object is hunter2"
+# ~~~
+#
+# You can optionally pass a block to unwrap in order to limit the scope where the
+# unwrapped value is visible.
+#
+# ~~~puppet
+# $pw = Sensitive.new('hunter2')
+# notice("Wrapped object is $pw") #=> Prints "Wrapped object is Sensitive [value redacted]"
+# $pw.unwrap |$unwrapped| {
+#   notice("Unwrapped object is $unwrapped") #=> Prints "Unwrapped object is hunter2"
+# }
+# # $unwrapped is now out of scope
+# ~~~
+#
+#
+# @since 4.0.0
+#
+Puppet::Functions.create_function(:unwrap) do
+  dispatch :unwrap do
+    param 'Sensitive', :arg
+    optional_block_param
+  end
+
+  def unwrap(arg)
+    unwrapped = arg.unwrap
+    if block_given?
+      yield(unwrapped)
```

Comment:
FWIW, returning the result of the block seems the more intuitive behavior to me. Being able to pass in a block that keeps the unwrapped value "contained" so I can do a simple computation on it and then return the result of that to the outside world, without having to bring out the unwrapped value, feels both safer from the perspective of sensitive data handling, and also cleaner from a functional-ish perspective